src/CommunityToolkit.Aspire.Hosting.Ollama/OllamaResourceBuilderExtensions.cs

```diff
@@ -55,63 +51,76 @@ public static IResourceBuilder<OllamaResource> WithDataVolume(this IResourceBuil
+    /// <param name="modelName">The name of the LLM from Hugging Face to download on initial startup.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    /// <remarks>
-    /// This method adds the connection string and model references to the application model.
-    /// </remarks>
-    public static IResourceBuilder<T> WithReference<T>(this IResourceBuilder<T> builder, IResourceBuilder<OllamaResource> ollama) where T : IResourceWithEnvironment
+    public static IResourceBuilder<OllamaModelResource> AddHuggingFaceModel(this IResourceBuilder<OllamaResource> builder, string name, string modelName)
     {
         ArgumentNullException.ThrowIfNull(builder, nameof(builder));
-        ArgumentNullException.ThrowIfNull(ollama, nameof(ollama));
+        ArgumentException.ThrowIfNullOrWhiteSpace(modelName, nameof(modelName));
+
+        builder.ApplicationBuilder.Services.TryAddLifecycleHook<OllamaModelResourceLifecycleHook>();
 
-        var resource = (IResourceWithConnectionString)ollama.Resource;
-        return builder.WithEnvironment(context =>
+        if (!modelName.StartsWith("hf.co/") && !modelName.StartsWith("huggingface.co/"))
         {
-            var connectionStringName = resource.ConnectionStringEnvironmentVariable ?? $"{ConnectionStringEnvironmentName}{resource.Name}";
+            modelName = "hf.co/" + modelName;
+        }
+
+        builder.Resource.AddModel(modelName);
+        var modelResource = new OllamaModelResource(name, modelName, builder.Resource);
 
-            context.EnvironmentVariables[connectionStringName] = new ConnectionStringReference(resource, optional: false);
+        return builder.ApplicationBuilder.AddResource(modelResource);
+    }
+
+    /// <summary>
+    /// Adds and sets the default model to be configured on the Ollama server.
+    /// </summary>
+    /// <param name="builder">The <see cref="IResourceBuilder{T}"/>.</param>
+    /// <param name="modelName">The name of the model.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<OllamaResource> WithDefaultModel(this IResourceBuilder<OllamaResource> builder, string modelName)
+    {
+        ArgumentNullException.ThrowIfNull(builder, nameof(builder));
+        ArgumentException.ThrowIfNullOrWhiteSpace(modelName, nameof(modelName));
 
-            for (int i = 0; i < ollama.Resource.Models.Count; i++)
-            {
-                var model = ollama.Resource.Models[i];
-                context.EnvironmentVariables[$"Aspire__OllamaSharp__{resource.Name}__Models__{i}"] = model;
+        builder.AddModel(modelName);
 
-                if (model == ollama.Resource.DefaultModel)
-                {
-                    context.EnvironmentVariables[$"Aspire__OllamaSharp__{resource.Name}__SelectedModel"] = model;
-                }
-            }
-        });
+        builder.Resource.SetDefaultModel(modelName);
+        return builder;
```

Comment:
As discussed, let's remove this method and instead you'll use the `OllamaModelResource` as the thing you pass in if you want a default model specified.